### ISSUE TITLE
Unknown compiler option 'jsxImportSource' issue..

### ISSUE THREAD

**Issue by leekeunhwan** [pedronauck/docz#1642]
## Question

Hi

I want to use docz

Unknown compiler option 'jsxImportSource' and
same issue occurs

As you use twin.macro in your project, in tsconfig 'jsxImportSource'
I am using it by setting it to '@emotion/react'.

The above issue occurred and the build is not working, is there any solution?

I tried to change appTsConfig in doczrc, but it doesn't work.

I am also wondering if there is a way to operate it with a separate tsconfig or if there is a plug-in that can solve it.

![스크린샷 2021-06-29 오후 2 38 41](https://user-images.githubusercontent.com/35440414/123743099-b45d7200-d8e7-11eb-8c3c-ad24250f9028.png)
![스크린샷 2021-06-29 오후 2 38 58](https://user-images.githubusercontent.com/35440414/123743130-c0493400-d8e7-11eb-87a5-647d37c7c162.png)

**Comment by adbayb** [comment]
Same with the `importsNotUsedAsValues` TS config 😅 .

It seems the root cause comes from the fact the TypeScript runtime is specified as a direct dependency inside the [docz-core package](https://github.com/pedronauck/docz/blob/main/core/docz-core/package.json#L53) and it seems to be pretty old (3.5.3 vs the latest 4.3.5: `jsxImportSource` was introduced in 4.1).

@pedronauck Could we imagine to have typescript specified as a peer dependency? The config is defined consumer side and depends on the TypeScript version specified by the consumer. It would sense to do so (and will also avoid multiple typescript copies in the project :) )

**Comment by leekeunhwan** [comment]
@pedronauck How are we going to solve this problem?

**Comment by renatobenks** [comment]
Thank you for the report guys! We're currently working to fully upgrade everything inside docz but we try can manage a quick fix to that earlier, and typescript is not a peer-dependency, we need it to build processes. But we'll figure a way out to fix that ASAP.

**Comment by adbayb** [comment]
@renatobenks Is it feasible to specify typescript as a devDependency? If yes, typescript will be available for your development needs (build process) without impacting consumer (dev dependencies are ignored while `npm install` consumer side).

**Comment by renatobenks** [comment]
Yes, totally! Let me check, because it should be as dev-dependency 🤔

**Comment by renatobenks** [comment]
However @adbayb, `npm install` actually install dev dependencies as well, as long as it's not with the `—-prod` within

**Comment by adbayb** [comment]
@renatobenks npm (or whatever node package manager) will install `devDependencies` of a given package when the `package.json` of this package is available locally (unless you set the `--prod` flag as you mentioned). 
**However** if the package is retrieved from a remote location (npm registry, git repository...), the prod flag is implicitly set (ie. no need to set it consumer side) and npm will only install the package `dependencies` (ignoring its `devDependencies`). 

So, here:
- If you run `yarn install` in the docz monorepo (local/dev environment) => local package case => `devDependencies`/`dependencies` will be installed
- If a consumer runs `yarn add docz` in their application => remote location case => `dependencies` will be installed only

**Comment by renatobenks** [comment]
Oh, what I see what you mean now, sorry! 😅

Sure thing, can you send a PR and see what we got?

**Comment by renatobenks** [comment]
I'll be working on a few issues on docz today, so let me know.

**Comment by adbayb** [comment]
@renatobenks Unfortunately not today but I can try to book some time tomorrow

**Comment by renatobenks** [comment]
Awesome! I just advice you to make sure that both rollup and lerna will keep working as properly because maybe, just maybe, that's why typescript is not a dev-dependency as it should.

**Comment by adbayb** [comment]
@renatobenks To let you know, I'm [on it](https://github.com/adbayb/docz/tree/fix/dependencies).

**Comment by adbayb** [comment]
@renatobenks and @pedronauck : done ✅ Review needed https://github.com/doczjs/docz/pull/1647 🙏
